src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1256,17 +1260,28 @@ private void PreparePlainExecutables()
             exe.Annotate(CustomResource.OtelServiceInstanceIdAnnotation, exeInstance.Suffix);
             exe.Annotate(CustomResource.ResourceNameAnnotation, executable.Name);
 
-            var supportedLaunchConfigurations = ExtensionUtils.GetSupportedLaunchConfigurations(_configuration);
-
-            if (executable.TryGetLastAnnotation<SupportsDebuggingAnnotation>(out var supportsDebuggingAnnotation)
-                && !string.IsNullOrEmpty(_configuration[DebugSessionPortVar])
-                && supportedLaunchConfigurations is not null
-                && supportedLaunchConfigurations.Contains(supportsDebuggingAnnotation.LaunchConfigurationType))
+            var isProcessExecution = true;
+            if (executable.SupportsDebugging(_configuration, out var supportsDebuggingAnnotation))
             {
-                exe.Spec.ExecutionType = ExecutionType.IDE;
-                supportsDebuggingAnnotation.LaunchConfigurationAnnotator(exe, _configuration[KnownConfigNames.DebugSessionRunMode] ?? ExecutableLaunchMode.NoDebug);
+                var launchConfigurationProducerOptions = new LaunchConfigurationProducerOptions
+                {
+                    DebugConsoleLogger = _backchannelLoggerProvider.CreateLogger(executable.Name),
+                    Mode = _configuration[KnownConfigNames.DebugSessionRunMode] ?? ExecutableLaunchMode.NoDebug
+                };
+
+                try
+                {
+                    supportsDebuggingAnnotation.LaunchConfigurationAnnotator(exe, launchConfigurationProducerOptions);
+                    exe.Spec.ExecutionType = ExecutionType.IDE;
+                    isProcessExecution = false;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to configure debugging for executable resource '{ResourceName}'. Falling back to process execution.", executable.Name);
+                }
             }
-            else
+            
+            if (isProcessExecution)
```

Comment:
We should fall back to process execution for both project and plain executables if LaunchConfigurationAnnotator throws. This ensures that other resources in the app are runnable.